python/lsst/obs/base/ingest.py

```diff
@@ -423,6 +635,20 @@ def prep(self, files, *, pool: Optional[Pool] = None, processes: int = 1) -> Ite
             pool = Pool(processes)
         mapFunc = map if pool is None else pool.imap_unordered
 
+        # Look for index files and read them
+        # There should be far fewer index files than data files
+        index_entries, files, good_index_files, bad_index_files = self.expandIndexFiles(files)
+        if bad_index_files:
+            self.log.info("Failed to read the following index files:"),
+            for bad in sorted(bad_index_files):
+                self.log.info("- %s", bad)
+
+        indexFileData = self.processIndexEntries(index_entries)
+        if indexFileData:
+            self.log.info("Successfully extracted metadata for %d file%s from %d index file%s",
+                          len(indexFileData), "" if len(indexFileData) == 1 else "s",
+                          len(good_index_files), "" if len(good_index_files) == 1 else "s")
```

Comment:
I'm never a fan of these pluralization formatters, but I guess this is fine. I'm torn on whether this wants a comment (took me a bit to figure out what was going on) or lambda/closure.